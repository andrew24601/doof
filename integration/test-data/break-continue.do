// Test break and continue in loops

// Continue in for loop

// Break in for loop

// Continue in while loop

// Break in nested loop

function main(): int {
    let result = "";
    for (let i = 0; i < 10; i++) {
        if (i % 2 == 0) {
            continue;
        }
        result = result + i;
    }
    result = result + "|";
    for (let j = 0; j < 10; j++) {
        if (j > 3) {
            break;
        }
        result = result + j;
    }
    result = result + "|";
    let k = 0;
    while (k < 10) {
        k++;
        if (k % 3 == 0) {
            continue;
        }
        result = result + k;
    }
    result = result + "|";
    for (let outer = 0; outer < 3; outer++) {
        for (let inner = 0; inner < 3; inner++) {
            if (inner == 1) {
                break;
            }
            result = result + outer + inner;
        }
    }
    println(result);
    return 0;
}