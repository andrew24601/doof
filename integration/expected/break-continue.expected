13579|0123|12457810|001020